rllib/agents/es/es.py

```diff
@@ -33,32 +34,123 @@
+    """Defines an ESTrainer configuration class from which an ESTrainer can be built.
+
+    Example:
+        >>> config = ESConfig().training(sgd_stepsize=0.02, report_length=20)\
+        ...     .resources(num_gpus=0)\
+        ...     .rollouts(num_rollout_workers=4)
+        >>> print(config.to_dict())
+        >>> # Build a Trainer object from the config and run 1 training iteration.
+        >>> trainer = config.build(env="CartPole-v1")
+        >>> trainer.train()
+
+    Example:
+        >>> from ray import tune
+        >>> config = ESConfig()
+        >>> # Print out some default values.
+        >>> print(config.action_noise_std)
+        >>> # Update the config object.
+        >>> config.training(rollouts_used=tune.grid_search([32, 64]), eval_prob=0.5)
+        >>> # Set the config object's env.
+        >>> config.environment(env="CartPole-v1")
+        >>> # Use to_dict() to get the old-style python config dict
+        >>> # when running with tune.
+        >>> tune.run(
+        ...     "ES",
+        ...     stop={"episode_reward_mean": 200},
+        ...     config=config.to_dict(),
+        ... )
+    """
+
+    def __init__(self):
+        """Initializes a ESConfig instance."""
+        super().__init__(trainer_class=ESTrainer)
+
+        # fmt: off
+        # __sphinx_doc_begin__
+
+        # ES specific settings:
+        self.action_noise_std = 0.01
+        self.l2_coeff = 0.005
+        self.noise_stdev = 0.02
+        self.episodes_per_batch = 1000
+        self.eval_prob = 0.03
+        self.return_proc_mode = "centered_rank"
+        self.stepsize = 0.01
+        self.noise_size = 250000000
+        self.report_length = 10
+
+        # Override some of TrainerConfig's default values with ES-specific values.
+        self.train_batch_size = 10000
+        self.num_workers = 10
+        self.observation_filter = "MeanStdFilter"
+        # ARS will use Trainer's evaluation WorkerSet (if evaluation_interval > 0).
+        # Therefore, we must be careful not to use more than 1 env per eval worker
+        # (would break ARSPolicy's compute_single_action method) and to not do
+        # obs-filtering.
+        self.evaluation_config["num_envs_per_worker"] = 1
+        self.evaluation_config["observation_filter"] = "NoFilter"
+
+        # __sphinx_doc_end__
+        # fmt: on
+
+    @override(TrainerConfig)
+    def training(
+        self,
+        *,
+        action_noise_std: Optional[float] = None,
+        l2_coeff: Optional[float] = None,
+        noise_stdev: Optional[int] = None,
+        episodes_per_batch: Optional[int] = None,
+        eval_prob: Optional[float] = None,
+        return_proc_mode: Optional[int] = None,
+        stepsize: Optional[float] = None,
+        noise_size: Optional[int] = None,
+        report_length: Optional[int] = None,
+        **kwargs,
+    ) -> "ESConfig":
+        """Sets the training related configuration.
+
+        Args:
+            action_noise_std:
+            l2_coeff:
```

Comment:
documentation seems to be missing for these parameters - I think we need those!

Reply:
Yeah, it never existed in the first place :/ Will try to find out what these do exactly. ...